ahadb802/Webpack-Practice
# Testing To Do list: part 1

## Testing To Do list: part 1

- Use jest framework for testing.
- Create a test file `([..].test.js) `for a file containing the **add item and delete item** functions that you want to test.
- Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:
- Mock a storage object to "imitate" localStorage operations (only if you are using jest `< 24.x,` in newer versions localStorage is mocked automatically).
- Mock HTML to test if add/delete functions add or remove exactly one `<li> `element to/from the list in the DOM.
- Make sure you group your tests using the `describe()` method.
